gazelle/docs/installation_and_usage.md

```diff
@@ -0,0 +1,156 @@
+# Installation and Usage
+
+## Example
+
+Examples of using Gazelle with Python can be found in the `rules_python`
+repo:
+
+* bzlmod: {gh-path}`examples/bzlmod_build_file_generation`
+* WORKSPACE: {gh-path}`examples/build_file_generation`
+
+:::{note}
+The following documentation covers using bzlmod.
+:::
+
+
+## Adding Gazelle to your project
+
+First, you'll need to add Gazelle to your `MODULE.bazel` file. Get the current
+version of Gazelle from their [releases page][gazelle-releases].
+
+[gazelle-releases]: https://github.com/bazel-contrib/bazel-gazelle/releases/
+
+See the installation `MODULE.bazel` snippet on the `rules_python`
+[releases page][rules-python-releases] in order to configure `rules_python`.
+
+[rules-python-releases]: https://github.com/bazel-contrib/rules_python/releases
```

Comment:
I would like to point users to the BCR: https://registry.bazel.build/modules/rules_python_gazelle_plugin

If the instructions are not up-to-date there (but I think they are), then maybe we can clean that up as well?

EDIT: it's just a move, it is fine to leave this as is and make changes in the upcoming PRs.